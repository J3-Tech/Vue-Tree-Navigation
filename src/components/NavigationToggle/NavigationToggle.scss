.navigation-toggle {
  position: relative;
  top:      -3px;
  padding:  6px;
  cursor:   pointer;

  .navigation-toggle-icon {
    display:      inline-block;
    padding:      3px;
    border:       solid #000;
    border-width: 0 2px 2px 0;
    transform:    rotate(45deg);
  }

  &.closed {
    .navigation-toggle-icon {
      transform: rotate(-45deg);
    }
  }
}
