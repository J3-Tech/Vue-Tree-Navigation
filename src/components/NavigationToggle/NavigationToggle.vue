<template>
  <span
    class="NavigationToggle"
    :class="{ 'NavigationToggle--closed': !this.open }">
    <span
      class="NavigationToggle__icon"
      :class="{ 'NavigationToggle__icon--closed': !this.open }"></span>
  </span>
</template>

<script>
export default {
  props: {
    open: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
};
</script>

<style lang="scss">
@import './NavigationToggle.scss';
</style>
