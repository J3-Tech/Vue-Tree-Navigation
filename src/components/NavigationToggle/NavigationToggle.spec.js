import { mount } from 'vue-test-utils';

import NavigationToggle from './NavigationToggle';

describe('NavigationToggle ', () => {
  it('renders', () => {
    const wrapper = mount(NavigationToggle, {
      propsData: {
        isClosed: true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  context('with `isClosed` property equal to true', () => {
    it('assigns `closed` class to outer span', () => {
      const wrapper = mount(NavigationToggle, {
        propsData: {
          isClosed: true,
        },
      });

      expect(wrapper.classes()).toContain('closed');
    });
  });

  context('with `isClosed` property equal to false', () => {
    const wrapper = mount(NavigationToggle, {
      propsData: {
        isClosed: false,
      },
    });

    expect(wrapper.classes()).not.toContain('closed');
  });
});
