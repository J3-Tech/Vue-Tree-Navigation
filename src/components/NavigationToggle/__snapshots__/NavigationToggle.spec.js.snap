// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationToggle  renders 1`] = `<span class="navigation-toggle closed"><span class="navigation-toggle-icon"></span></span>`;
